src/tab/Tab.tsx

```diff
@@ -16,20 +16,26 @@ export interface TabProps {
   items: TabItem[];
   children: React.ReactNode[];
   testid?: string;
-  initialActiveTabIndex?: number;
+  activeTabIndex?: number;
   customClassName?: string;
+  onTabChange?: (index: number) => void;
 }
 
 function Tab({
   testid,
   items,
-  initialActiveTabIndex = 0,
+  activeTabIndex: activeTabIndexFromProps = 0,
   children,
-  customClassName
+  customClassName,
+  onTabChange
 }: TabProps) {
-  const [activeTabIndex, setActiveTabIndex] = useState(initialActiveTabIndex);
+  const [activeTabIndex, setActiveTabIndex] = useState(activeTabIndexFromProps);
   const tabClassName = classNames("tab", customClassName);
 
+  useEffect(() => {
+    setActiveTabIndex(activeTabIndexFromProps);
+  }, [activeTabIndexFromProps]);
```

Comment:
If we change the implementation to satisfy the requirements I described in my previous comment, I think we will be able to remove this useEffect block altogether.